### DISCUSSION TITLE
"Automatic field wrapping inside DIV" breaks my CSS

### DISCUSSION THREAD

**Discussion by folknor** [TarekRaafat/autoComplete.js#227] [Q&A] [answered]
Can you explain the reason for wrapping the input field in a div? It breaks my CSS, which is based on https://getbootstrap.com/docs/5.0/forms/floating-labels/#example

I can fix it, but why did you break it?

**Comment by TarekRaafat** [comment, marked as answer]
The `wrapper` was not meant to break anything as much as to improve the list rendering and the WAI-ARIA support.

Yet, I was already planing to make it optional, and I'll make sure to avail that API the soonest.

**Reply by folknor** [reply]
I did not think you meant to break anything :-)

The changelog/release notes and git commits contain very little information (explanation), I was just curious what the reason was.

Thank you!

**Reply by TarekRaafat** [reply]
No worries, I understand, and I'll make sure to include enough information in the future.

You can check the `wrapper` API switch during development under the [dev](https://github.com/TarekRaafat/autoComplete.js/tree/dev) branch.

**Reply by TarekRaafat** [reply]
Hello @folknor, [v10.1.0](https://github.com/TarekRaafat/autoComplete.js/releases/tag/v10.1.0) is released with the `wrapper` API.

Please check it and let me how it goes.

**Reply by folknor** [reply]
Excellent, works just like before with `wrapper: false`. Thanks!

I have a weird issue now though that after I select an item in the results, the autocomplete container is never shown again.

I'm not sure if it's a bug in my code or in the library, but I believe it's in the library.

`resultsList.element(element, data)` is invoked, and `const { input, query, matches, results } = data` has the correct data for the new search, but then the list just simply isn't shown.

**Reply by TarekRaafat** [reply]
Do you get that behavior with the `wrapper` false or true?

**Reply by folknor** [reply]
> Do you get that behavior with the `wrapper` false or true?

Both. I was going to do some testing immediately and file an issue but was interrupted. It must be some problem in my code.

**Reply by folknor** [reply]
The problem only comes when I have a custom `resultsList.id` set.

**Reply by folknor** [reply]
Ah I see. In a previous version of autoComplete.js I had to manually remove the rendered resultsList HTML because I had to override the click event handler.

Now I don't need to any more, but I was still nuking the HTML, and autoComplete.js no longer recreates it.
